target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-sxtsq-5-ac.dts

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/* Copyright (c) 2020, Robert Marko <robimarko@gmail.com> */
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	model = "MikroTik SXTsq 5 ac (RBSXTsqG-5acD)";
+	compatible = "mikrotik,sxtsq-5-ac";
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x10000000>;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-running = &led_user;
+		led-upgrade = &led_user;
+	};
+
+	soc {
+		rng@22000 {
+			status = "okay";
+		};
+
+		mdio@90000 {
+			status = "okay";
+		};
+
+		counter@4a1000 {
+			compatible = "qcom,qca-gcnt";
+			reg = <0x4a1000 0x4>;
+		};
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+			switch_lan_bmp = <0x20>;
+			switch_wan_bmp = <0x00>;
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+		edma@c080000 {
+			qcom,num_gmac = <1>;
+			status = "okay";
```

Comment:
Could it be that `qcom,single-phy;` is missing here?

Reply:
Adding `qcom,single-phy` makes the `eth0` interface not to be loaded, with this kernel error:
```
[    1.691062] [<c0626ba8>] (driver_register) from [<c0302764>] (do_one_initcall+0x84/0x1d8)
[    1.699305] [<c0302764>] (do_one_initcall) from [<c0b00fcc>] (kernel_init_freeable+0x1c4/0x26c)
[    1.707381] [<c0b00fcc>] (kernel_init_freeable) from [<c086941c>] (kernel_init+0x8/0x114)
[    1.715884] [<c086941c>] (kernel_init) from [<c03010e8>] (ret_from_fork+0x14/0x2c)
[    1.724211] Exception stack(0xcf82bfb0 to 0xcf82bff8)
[    1.731677] bfa0:                                     00000000 00000000 00000000 00000000
[    1.736805] bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    1.744961] bfe0: 00000000 00000000 00000000 00000000 00000013 00000000
[    1.753161] ---[ end trace eca84138885b5897 ]---
[    1.759814] ------------[ cut here ]------------
[    1.764411] WARNING: CPU: 2 PID: 1 at mm/vmalloc.c:2244 __vunmap+0x25c/0x260
[    1.769002] Trying to vfree() nonexistent vm area ((ptrval))
[    1.776090] Modules linked in:
[    1.781697] CPU: 2 PID: 1 Comm: swapper/0 Tainted: G        W         5.4.89 #0
[    1.784539] Hardware name: Generic DT based system
[    1.791758] [<c030e544>] (unwind_backtrace) from [<c030afb4>] (show_stack+0x10/0x14)
[    1.796618] [<c030afb4>] (show_stack) from [<c08536b8>] (dump_stack+0x94/0xa8)
[    1.804514] [<c08536b8>] (dump_stack) from [<c03213ac>] (__warn+0xbc/0xd8)
[    1.811541] [<c03213ac>] (__warn) from [<c032144c>] (warn_slowpath_fmt+0x84/0x94)
[    1.818402] [<c032144c>] (warn_slowpath_fmt) from [<c041bcdc>] (__vunmap+0x25c/0x260)
[    1.825957] [<c041bcdc>] (__vunmap) from [<c06299c4>] (release_nodes+0x1ac/0x1d8)
[    1.833769] [<c06299c4>] (release_nodes) from [<c0625f98>] (really_probe+0x114/0x358)
[    1.841234] [<c0625f98>] (really_probe) from [<c06265c0>] (device_driver_attach+0x58/0x60)
[    1.849046] [<c06265c0>] (device_driver_attach) from [<c0626620>] (__driver_attach+0x58/0xcc)
[    1.857205] [<c0626620>] (__driver_attach) from [<c06245c8>] (bus_for_each_dev+0x68/0x8c)
[    1.865800] [<c06245c8>] (bus_for_each_dev) from [<c062554c>] (bus_add_driver+0x1c8/0x1d8)
[    1.873960] [<c062554c>] (bus_add_driver) from [<c0626ba8>] (driver_register+0x74/0x108)
[    1.882120] [<c0626ba8>] (driver_register) from [<c0302764>] (do_one_initcall+0x84/0x1d8)
[    1.890367] [<c0302764>] (do_one_initcall) from [<c0b00fcc>] (kernel_init_freeable+0x1c4/0x26c)
[    1.898440] [<c0b00fcc>] (kernel_init_freeable) from [<c086941c>] (kernel_init+0x8/0x114)
[    1.906947] [<c086941c>] (kernel_init) from [<c03010e8>] (ret_from_fork+0x14/0x2c)
[    1.915270] Exception stack(0xcf82bfb0 to 0xcf82bff8)
[    1.922735] bfa0:                                     00000000 00000000 00000000 00000000
[    1.927863] bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
[    1.936021] bfe0: 00000000 00000000 00000000 00000000 00000013 00000000
[    1.944211] ---[ end trace eca84138885b5898 ]---
[    1.950640] ess_edma: probe of c080000.edma failed with error -2
```
whereas, without it, it was correctly loaded:
```
[    2.852076] ess_edma c080000.edma eth0: Link is Down
[    6.011935] ess_edma c080000.edma eth0: Link is Up - 100Mbps/Full - flow control off
[    6.012879] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    7.088016] ess_edma c080000.edma eth0: Link is Down
```


Therefore, either `qcom,single-phy` is not missing, or it must be accompanied by something else...

Reply:
Whups sry. Yeah I just guessed why vlans are not working. As @spolack alredy mentioned has the device the switch enabled. His workarounds are working. :)

Reply:
No worries. There are a few devices in the ipq40xx target with the feature. I tried using their configurations on the SXT and the result was the same: no eth0 and kernel error. :disappointed: